pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -105,6 +105,26 @@ type OperationExecutor interface {
 	// actual state of the world to reflect that.
 	UnmountDevice(deviceToDetach AttachedVolume, actualStateOfWorld ActualStateOfWorldMounterUpdater, mounter mount.Interface) error
 
+	// MapVolume is used when the volumeMode is 'Block'. This method creates a symbolic link to the block device on global map path.
+	// Specifically it will:
+	// * Wait for the device to finish attaching (for attachable volumes only).
+	// * Map volume to global map path using symbolic link (for attachable volumes only).
+	// * Update actual state of world to reflect volume is globally mounted (for
+	//   attachable volumes only).
+	// * Map the volume to the pod specific path using symbolic link.
+	// * Update actual state of world to reflect volume is mounted to the pod
+	//   path.
+	MapVolume(waitForAttachTimeout time.Duration, volumeToMount VolumeToMount, actualStateOfWorld ActualStateOfWorldMounterUpdater) error
+
+	// UnmapVolume unmaps symbolic link to the volume from both the pod specified in
+	// volumeToUnmount and global map path. And then, updates the actual state of the world to reflect that.
+	UnmapVolume(volumeToUnmount MountedVolume, actualStateOfWorld ActualStateOfWorldMounterUpdater) error
+
+	// UnmapDevice unmaps the volumes global map path from the device (for
```

Comment:
IIUC, it doesn't actually unmap anything, it just checks if all references to the device have been removed?